lib/screens/v2/candidate_generator/dup/departures.ex

```diff
@@ -525,9 +526,24 @@ defmodule Screens.V2.CandidateGenerator.Dup.Departures do
     # Routes not in overnight mode will be nil. Can ignore those.
     |> Enum.reject(&is_nil/1)
     |> Enum.sort_by(fn %Departure{schedule: schedule} -> schedule.departure_time end)
+    |> Enum.split_with(fn
+      # Split up the upcoming departures for each route into a tuple of lists, {todays_schedules, overnight_schedules}
+      %{schedule: %{departure_time: nil}} ->
```

Comment:
* We should use the structs for pattern-matching to enable better compile-time checks and code analysis, e.g. `%Departure{schedule: %Schedule{departure_time: nil}}`.
* The flag for producing the overnight icon is _both_ departure and arrival time being `nil`. One or the other could potentially be `nil` for an ordinary, "real" departure; we might filter out "arrival-only" departures at an earlier stage of things anyway, but best to align things as closely as possible in case that ever changes.